src/types/topology/routes.rs

```diff
@@ -0,0 +1,10 @@
+use crate::types::topology::{NodeId, NodeRef};
+use std::collections::HashMap;
+
+pub struct Route {
+    //route is the actual vec of references, path is just a "treasure map" with ids
+    path: Vec<NodeRef>,
+}
+pub type Routes = HashMap<NodeId, Route>;
```

Comment:
```suggestion
```
Why we have an hashmap of routes wouldn't it be better using a vec? Also I don't see the gains in using a type in this case i would just remove it. Expecially because it can cause confusion.